### ISSUE TITLE
NotifyUsersWorkflowAction causes exception: "the method 'scalarvalueonly' does not exist on 'ArrayData'"

### ISSUE THREAD

**Issue by guttmann** [symbiote/silverstripe-advancedworkflow#391]
I'm having issues with the advancedworkflow module (3.8.2) in a SilverStripe 3.6.7 website.

I believe the security vulnerability fix for [ss-2018-021](https://www.silverstripe.org/download/security-releases/ss-2018-021) has caused an issue
in the NotifyUsersWorkflowAction, specifically these lines: 

https://github.com/symbiote/silverstripe-advancedworkflow/blob/3.8.2/code/actions/NotifyUsersWorkflowAction.php#L108-L110

The error I receive is:

```
ERROR [User Error]: Uncaught Exception: Object->__call(): the method 'scalarvalueonly' does not exist on 'ArrayData', or the method is not public.
IN POST /admin/pages/edit/EditForm/3022
Line 780 in framework/core/Object.php

Trace
=====
Object->__call(scalarvalueonly,Array)
DataObject.php:2677

DataObject->setField(Assignee,ArrayData)
ViewableData.php:142

ViewableData->__set(Assignee,ArrayData)
ViewableData.php:720

ViewableData_Customised->__set(Assignee,ArrayData)
NotifyUsersWorkflowAction.php:109

NotifyUsersWorkflowAction->execute(WorkflowInstance)
WorkflowInstance.php:285

WorkflowInstance->execute()
WorkflowInstance.php:366

WorkflowInstance->performTransition(WorkflowTransition)
WorkflowInstance.php:301

WorkflowInstance->execute()
WorkflowInstance.php:366

WorkflowInstance->performTransition(WorkflowTransition)
WorkflowService.php:211

WorkflowService->executeTransition(Page,65)
AdvancedWorkflowExtension.php:119

AdvancedWorkflowExtension->updateworkflow(Array,CMSForm,SS_HTTPRequest)
Object.php:749

Object->__call(updateworkflow,Array)
Form.php:466

Form->httpSubmission(SS_HTTPRequest)
RequestHandler.php:288

RequestHandler->handleAction(SS_HTTPRequest,httpSubmission)
RequestHandler.php:200

RequestHandler->handleRequest(SS_HTTPRequest,DataModel)
RequestHandler.php:222

RequestHandler->handleRequest(SS_HTTPRequest,DataModel)
Controller.php:158

Controller->handleRequest(SS_HTTPRequest,DataModel)
LeftAndMain.php:464

LeftAndMain->handleRequest(SS_HTTPRequest,DataModel)
AdminRootController.php:92

AdminRootController->handleRequest(SS_HTTPRequest,DataModel)
Director.php:385

Director::handleRequest(SS_HTTPRequest,Session,DataModel)
Director.php:149

Director::direct(/admin/pages/edit/EditForm/3022,DataModel)
main.php:208
```

I believe the fix is to change the line 109 from:

```
$item->Assignee = new ArrayData($assigneeVars);
```

to

```
$item->customise(['Assignee' => new ArrayData($assigneeVars)]);
```

But - I've noticed that this also appears to affect master, 4.x, and 5.x so not sure the best way to suggest a fix to all of them.

(example of the same issue in master: https://github.com/symbiote/silverstripe-advancedworkflow/blob/master/src/Actions/NotifyUsersWorkflowAction.php#L145-L147)

Could someone advise the best way for me to approach providing a fix for all versions?

**Comment by guttmann** [comment]
I've logged an issue in silverstripe/silverstripe-framework that I think is the root cause of this issue: silverstripe/silverstripe-framework#8883

**Comment by robbieaverill** [comment]
Fixed by merging #392, thanks @gutt
